File: backend/internal/dto/emergency_fund.go
package dto

type EFSummaryResponse struct {
	TotalEmergencyFund MoneyValue       `json:"total_emergency_fund"`
	MonthlyLivingCost  MoneyValue       `json:"monthly_living_cost"`
	TargetMonths       int              `json:"target_months"`
	TargetAmount       MoneyValue       `json:"target_amount"`
	CoverageMonths     float64          `json:"coverage_months"`
	ProgressPercentage float64          `json:"progress_percentage"`
	Status             string           `json:"status"` // Aman, Kurang, Kritis
	TargetRationale    string           `json:"target_rationale"`
	DataSufficiency    *DataSufficiency `json:"data_sufficiency,omitempty"`
}

type UpdateEFConfigRequest struct {
	TargetMonths              int      `json:"target_months" binding:"required,min=1"`
	MonthlyLivingCostOverride *float64 `json:"monthly_living_cost_override"`
}

File: frontend/src/services/emergencyFund.ts
import api from '../utils/api';
import type { MoneyValue } from './dashboard';

export interface EFSummaryResponse {
  total_emergency_fund: MoneyValue;
  monthly_living_cost: MoneyValue;
  target_months: number;
  target_amount: MoneyValue;
  coverage_months: number;
  progress_percentage: number;
  status: 'Aman' | 'Kurang' | 'Kritis';
  target_rationale?: string;
  as_of?: string;
  formula_version?: string;
  assumptions?: string[];
  data_sufficiency?: {
    is_sufficient: boolean;
    missing_fields?: string[];
    uses_fallback_values?: boolean;
    confidence?: string;
  };
}

export interface UpdateEFConfigRequest {
  target_months: number;
  monthly_living_cost_override?: number | null;
}

export const emergencyFundService = {
  async getEFSummary(): Promise<EFSummaryResponse> {
    const res = await api.get('/emergency-fund/summary');
    return res.data.data;
  },

  async updateEFConfig(req: UpdateEFConfigRequest): Promise<void> {
    await api.put('/emergency-fund/config', req);
  },
};
export default emergencyFundService;
